### PR TITLE
quant docs: clean up hardswish

### PR DESCRIPTION
**Pull request by vkuzo**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #40377 quant docs: add and clean up ELU
* #40346 quant docs: add and clean up BatchNorm{n}d
* #40345 quant docs: add and clean up InstanceNorm{n}d
* #40343 quant docs: add and clean up GroupNorm
* #40342 quant docs: add and clean up LayerNorm
* #40341 quant docs: add and clean up hardtanh
* #40340 quant docs: add and clean up hardsigmoid
* **#40323 quant docs: clean up hardswish**

Cleans up the naming and the function param docs for quantized hardswish.
Remove redundant docstrings and link to floating point modules instead.

Test plan:
* build the docs on Mac OS
* verify that every link works as expected

Differential Revision: [D22152638](https://our.internmc.facebook.com/intern/diff/D22152638)

### CODE REVIEW

**Review by jerryzh168 — APPROVED** [review]

File: torch/nn/quantized/functional.py [hunk 1]
```diff
@@ -427,26 +427,12 @@ def hardtanh(input, min_val=-1., max_val=1., inplace=False):
...
-
-    `Searching for MobileNetV3`_.
-
-    .. math::
-        \text{Hardswish}(x) = \begin{cases}
-            0 & \text{if~} x \le -3, \\
-            x & \text{if~} x \ge +3, \\
-            x^2/6 & \text{otherwise}
-        \end{cases}
+    r"""This is the quantized version of :meth:`torch.nn.functional.hardswish`.
```
Comment (jerryzh168) [inline torch/nn/quantized/functional.py:430]: probably should be :meth:`~torch.nn.functional.hardswish`

File: docs/source/quantization.rst [hunk 2]
```diff
@@ -193,7 +193,7 @@ Basic activations are supported.
 * :meth:`~torch.nn.functional.adaptive_avg_pool2d` - Adaptive average pooling
 * :meth:`~torch.nn.functional.avg_pool2d` - Average pooling
 * :meth:`~torch.nn.functional.interpolate` - Interpolation
-* :meth:`~torch.nn.functional.hardswish` - Hard Swish
+* :meth:`~torch.nn.functional.hardswish` - Hardswish
```
Comment (jerryzh168) [inline docs/source/quantization.rst:196]: actually why are these method? I feel it makes more sense for these to be `func`

File: torch/nn/quantized/functional.py (hunk 1 above)
Comment (vkuzo) [reply torch/nn/quantized/functional.py]: it works and is used in pytorch codebase both ways, but sure

File: docs/source/quantization.rst (hunk 2 above)
Comment (vkuzo) [reply docs/source/quantization.rst:196]: not sure if there is any difference in the generated docs.  If this is important, might be good to be a separate PR on the entire file, so that it's consistent within the file

Comment (jerryzh168) [reply docs/source/quantization.rst:196]: yeah, it should be changed in a separate PR I think

**Review by raghuramank100 — APPROVED** [review]
